### PR TITLE
React: New task and contact forms

### PR DESCRIPTION
**Pull request by groshenkovamarina**

### CODE REVIEW

File: packages/react/src/components/form-photo-uploader/FormPhotoUploader.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import { FileUploader } from 'devextreme-react/file-uploader';
+
+export const FormPhotoUploader = () => {
+  const [isDropZoneActive, setDropZoneActive] = useState(false);
+
+  const onDropZoneEnter = useCallback((e) => {
+    if (e.dropZoneElement.id === 'uploader') {
+      setDropZoneActive(true);
+    }
+  }, [isDropZoneActive]);
```
Comment (const314) [inline packages/react/src/components/form-photo-uploader/FormPhotoUploader.tsx:14]: These two hooks will change every time isDropZoneActive changes, it's unnecessary in this case. Pass the empty array instead.

File: packages/react/src/components/task-form/TaskFormDetails.tsx [hunk 2]
```diff
@@ -47,105 +46,112 @@ const renderStatusField = (data: string) => (
...
+      className={classNames({ 'plain-styled-form': true, 'view-mode': !editing })}
+      validationGroup='taskValidationGroup'
+    >
+      <SimpleItem visible={!data.text}>
+        <FormTextbox
+          label='Subject'
+          value={data.text}
+          isEditing={!editing}
+          onValueChange={updateField('text')}
+        ></FormTextbox>
```
Comment (const314) [inline packages/react/src/components/task-form/TaskFormDetails.tsx:66]: Use self-closing tags for components without children

File: packages/react/src/pages/crm-contact-list/contact-new-form/contactNewForm.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+import Form, { Item as FormItem, ColCountByScreen } from 'devextreme-react/form';
+import { newContact } from '../../../shared/types/crm-contact';
+import { FormTextbox, FormPhotoUploader } from '../../../components';
+
+export const ContactNewForm = () => {
+  const validationGroup = 'contactNewValidationGroup';
+
+  const updateField = (field: string) => (value) => {
+    return { ...newContact, ...{ [field]: value } };
```
Comment (const314) [inline packages/react/src/pages/crm-contact-list/contact-new-form/contactNewForm.tsx:12]: Why not store newContact in the state?

File: packages/react/src/components/form-photo-uploader/FormPhotoUploader.tsx (hunk 1 above)
Comment (groshenkovamarina) [reply packages/react/src/components/form-photo-uploader/FormPhotoUploader.tsx]: Fix

File: packages/react/src/components/task-form/TaskFormDetails.tsx (hunk 2 above)
Comment (groshenkovamarina) [reply packages/react/src/components/task-form/TaskFormDetails.tsx]: Fixed

File: packages/react/src/pages/crm-contact-list/contact-new-form/contactNewForm.tsx (hunk 3 above)
Comment (groshenkovamarina) [reply packages/react/src/pages/crm-contact-list/contact-new-form/contactNewForm.tsx]: Fixd

File: packages/react/src/components/task-form/TaskFormDetails.tsx (hunk 2 above)
Comment (groshenkovamarina) [reply packages/react/src/components/task-form/TaskFormDetails.tsx]: Fixed

**Review by const314 — DISMISSED** [review]

File: packages/react/src/pages/planning-task-list/planning-task-list.tsx [hunk 4]
```diff
@@ -30,15 +31,15 @@ export const PlanningTaskList = () => {
...
-  const navigate = useNavigate();
-
   const [listView, kanbanView, ganttView] = listsData;
 
   const [view, setView] = useState(listView);
   const [index, setIndex] = useState(0);
   const [gridData, setGridData] = useState([]);
   const [filteredData, setFilteredData] = useState([]);
   const [loading, setLoading] = useState(true);
+  const [newTaskData, setNewTaskData] = useState(newTask);
```
Comment (Krijovnick) [inline packages/react/src/pages/planning-task-list/planning-task-list.tsx:41]: Why in the planning list page form opened with some filled values, but in the contact list page they all empty. I think we should make they all empty and remove `newTask` variable.

File: packages/react/src/shared/types/crm-contact.ts [hunk 5]
```diff
@@ -10,21 +10,45 @@ interface State {
...
-  company: string,
-  city: string,
-  state: State,
-  phone: string,
-  email: string,
-  image: string,
-  zipCode: number,
-  tasks: Task[],
-  activities: Activities,
+    id: number | null,
```
Comment (Krijovnick) [inline packages/react/src/shared/types/crm-contact.ts:13]: id should not be null

File: packages/react/src/shared/types/task.ts [hunk 6]
```diff
@@ -7,7 +7,7 @@ export type TaskPriority = 'Low' | 'Normal' | 'High';
 export type TaskStatus = 'Open' | 'In Progress' | 'Deferred' | 'Completed';
 
 export interface Task {
-    id: number,
+    id?: number,
```
Comment (Krijovnick) [inline packages/react/src/shared/types/task.ts:10]: id can't be undefined

File: packages/react/src/shared/types/task.ts [hunk 7]
```diff
@@ -28,4 +28,21 @@ export interface TaskProp {
 }
 export interface PlanningProps {
     dataSource: Task[];
+    changePopupVisibility?: () => void;
 }
+
+export const newTask: Task = {
```
Comment (Krijovnick) [inline packages/react/src/shared/types/task.ts:34]: It should be removed or moved to constants

**Review by Krijovnick — CHANGES_REQUESTED** [review]

File: packages/react/src/pages/crm-contact-list/crm-contact-list.tsx [hunk 8]
```diff
@@ -78,18 +81,26 @@ export const CRMContactList = () => {
...
+    setPopupVisible(!popupvisible);
+  };
+
+  const addContact = useCallback(() => {
+    changePopupVisibility();
   }, []);
 
+  const onSavePopupClick = () => {
+    changePopupVisibility();
+  };
```
[the remark this answers is not in the document]
Comment (groshenkovamarina) [reply packages/react/src/pages/crm-contact-list/crm-contact-list.tsx:94]: This method need for validation on save click

File: packages/react/src/pages/crm-contact-list/contact-new-form/contactNewForm.tsx [hunk 9]
```diff
@@ -0,0 +1,105 @@
+
+import React, { useState } from 'react';
+
+import Form, { Item as FormItem, ColCountByScreen } from 'devextreme-react/form';
+import { newContact, Contact } from '../../../shared/types/crm-contact';
+import { FormTextbox, FormPhotoUploader } from '../../../components';
+
+export const ContactNewForm = () => {
+  const [newContactData, setNewContactData] = useState<Contact>(newContact);
```
[the remark this answers is not in the document]
Comment (groshenkovamarina) [reply packages/react/src/pages/crm-contact-list/contact-new-form/contactNewForm.tsx:9]: I think, It is correctly to use objects type of Task and Contact instead empty objects here but I moved  them to constants

File: packages/react/src/components/form-popup/FormPopup.tsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+
+import { Popup, ToolbarItem } from 'devextreme-react/popup';
+import { useScreenSize } from './../../utils/media-query';
+
+type PopupProps = {
+  title: string,
+  children,
```
[the remark this answers is not in the document]
Comment (groshenkovamarina) [reply packages/react/src/components/form-popup/FormPopup.tsx]: Fixed

File: packages/react/src/components/planning-kanban/list/List.tsx [hunk 11]
```diff
@@ -22,23 +21,26 @@ const boardMenuItems = [{
...
-  index: number;
-  tasks: Task[];
-  onTaskDragStart: (e: DragStartEvent) => void;
-  onTaskDrop: (e: ReorderEvent) => void;
+  title: string,
+  index: number,
+  tasks: Task[],
+  onTaskDragStart: (e: DragStartEvent) => void,
+  onTaskDrop: (e: ReorderEvent) => void,
+  changePopupVisibility,
```
[the remark this answers is not in the document]
Comment (groshenkovamarina) [reply packages/react/src/components/planning-kanban/list/List.tsx]: Fixed

File: packages/react/src/pages/crm-contact-list/crm-contact-list.tsx [hunk 12]
```diff
@@ -52,6 +54,7 @@ export const CRMContactList = () => {
   const [status, setStatus] = useState(filterStatusList[0]);
   const [gridData, setGridData] = useState([]);
   const [loading, setLoading] = useState(true);
+  const [popupvisible, setPopupVisible] = useState(false);
```
[the remark this answers is not in the document]
Comment (groshenkovamarina) [reply packages/react/src/pages/crm-contact-list/crm-contact-list.tsx]: Fixed

File: packages/react/src/pages/planning-task-list/planning-task-list.tsx [hunk 13]
```diff
@@ -30,15 +31,15 @@ export const PlanningTaskList = () => {
...
+  const [popupvisible, setPopupVisible] = useState(false);
```
[the remark this answers is not in the document]
Comment (groshenkovamarina) [reply packages/react/src/pages/planning-task-list/planning-task-list.tsx]: Fixed

File: packages/react/src/shared/types/crm-contact.ts [hunk 14]
```diff
@@ -10,21 +10,45 @@ interface State {
...
+    zipCode: number,
+    phone: string,
+    email: string,
+    image: string,
+    activities: Activities,
+    opportunities: Opportunities,
+    tasks: Task[],
 }
+
+export const newContact: Contact = {
```
[the remark this answers is not in the document]
Comment (groshenkovamarina) [reply packages/react/src/shared/types/crm-contact.ts]: Fixed

File: packages/react/src/shared/types/crm-contact.ts [hunk 15]
```diff
@@ -10,21 +10,45 @@ interface State {
...
+  id: null,
```
[the remark this answers is not in the document]
Comment (groshenkovamarina) [reply packages/react/src/shared/types/crm-contact.ts]: Fixed

File: packages/react/src/pages/crm-contact-list/contact-new-form/contactNewForm.tsx [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+            label='First Name'
+            value={newContact.firstName}
+            isEditing={false}
+            onValueChange={updateField('firstName')}
+          />
+        </FormItem>
+        <FormItem>
+          <FormTextbox
+            label='Last Name'
+            value={newContact.lastName}
```
[the remark this answers is not in the document]
Comment (groshenkovamarina) [reply packages/react/src/pages/crm-contact-list/contact-new-form/contactNewForm.tsx]: Fixed

File: packages/react/src/shared/types/crm-contact.ts (hunk 5 above)
Comment (groshenkovamarina) [reply packages/react/src/shared/types/crm-contact.ts]: Fixed

File: packages/react/src/shared/types/task.ts (hunk 6 above)
Comment (groshenkovamarina) [reply packages/react/src/shared/types/task.ts:10]: Fixed

File: packages/react/src/pages/crm-contact-list/contact-new-form/contactNewForm.tsx [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+  const updateField = (field: string) => (value) => {
+    return setNewContactData({ ...newContactData, ...{ [field]: value } });
+  };
+
+  return (
+    <Form
+      labelMode='floating'
+      className='plain-styled-form'
+    >
+      <FormItem itemType='group'>
```
[the remark this answers is not in the document]
Comment (groshenkovamarina) [reply packages/react/src/pages/crm-contact-list/contact-new-form/contactNewForm.tsx]: Fixed

**Review by Krijovnick — APPROVED** [review]

**Review by const314 — APPROVED** [review]

**Pull request merged**